server.js

```diff
@@ -3,36 +3,92 @@ import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+import booksData from './data/books.json'
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+const Book = new mongoose.model('Book', {
+  bookID: Number,
+  title: String,
+  authors: String,
+  average_rating: Number,
+  isbn: Number,
+  isbn13: Number,
+  language_code: String,
+  num_pages: Number,
+  ratings_count: Number,
+  text_reviews_count: Number
+});
+
+
+if (process.env.RESET_DATABASE) {
+  const populateDatabase = async () => {
+    await Book.deleteMany()
+    booksData.forEach(async item => {
+      await new Book(item).save()
+    })
+  }
+  populateDatabase()
+}
+
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: "Service unavailable at the moment" })
+  }
+})
+
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Hello world, welcome to this books API built with MongoDB')
+})
+
+app.get('/books', async (req, res) => {
+  const allBooks = await Book.find()
+  res.json(allBooks)
 })
 
-// Start the server
+app.get('/books/id/:bookID', async (req, res) => {
+  const bookByID = await Book.findOne({ bookID: req.params.bookID });
+  if (bookByID) {
+    res.json(bookByID);
+  } else {
+    res.status(404).json({ error: 'Book not found' });
+  };
+});
+
+app.get('/books/author/:author', async (req, res) => {
+  const author = req.params.author
+  const booksByAuthor = await Book.find({ authors: { "$regex": author, "$options": "i" } })
+  res.json(booksByAuthor)
+})
+
+app.get('/books/top-rated', async (req, res) => {
+  const topRatedBooks = await Book.find({ average_rating: { $gte: 4.5 } })
```

Comment:
I love that you choose to sort out the top-rated books. Good thought and very clean with the regex! 👏🏼